### PR TITLE
Review whole doc. and improve it where warranted

### LINKED ISSUE #11: Audit pages for typos
Most if not all pages have a few typos or grammatical errors from the readme.io export.

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
@starbelly, this might close (issue) #11, but since that one's so generic... 😄 I present my take on fixing stuff, tweaking stuff, and other oddities. I also ran `markdownlint` on top of the final result, so it's seems more consistent.

### CODE REVIEW

File: content/en/docs/configuration/dependencies.md [hunk 1]
```diff
@@ -121,72 +121,69 @@ $ rebar3 update
...
 
-Note that `_checkouts` is an override, this means that for it to work a `dep` or `plugin` entry in rebar.config _must_ exist.
+Note that `_checkouts` is an override, this means that for it to work a `dep` or `plugin` entry in `rebar.config` _must_ exist.
 
 ## Fetching Order
 
 For a regular dependency tree, such as:
 
-```
+```shell
```
Comment (ferd) [inline content/en/docs/configuration/dependencies.md:145]: Is this something the linter demanded? these are absolutely not shell sections, they're ascii art. There's no reason they would need syntax highlighting :/

File: content/en/docs/configuration/profiles.md [hunk 2]
```diff
@@ -95,9 +96,9 @@ The order of application of profiles is therefore:
 Profiles are therefore a composable way to specify configuration subsets in a contextual manner.
 
 {{% blocks/callout type="info" title="Locking Dependencies" %}}
- Only dependencies list at the top level of `rebar.config`, the `default` profile,  are saved to `rebar.lock`. Other dependencies will not get locked.
+Only dependencies listed at the top level of `rebar.config`, the `default` profile, are saved to `rebar.lock`. Other dependencies will not get locked.
 
-If someone wants to "lock for production" (meaning with production-related profiles), the answer is to keep the default profile and to use [releases](/docs/deployment/releases), which allow to produce compiled artifacts that can be reused at any time.
+If someone wants to "lock for production" (meaning with production-related profiles), the answer is to keep the default profile and to use [releases](/docs/deployment/releases), which allows producing compiled artifacts that can be reused at any time.
```
Comment (ferd) [inline content/en/docs/configuration/profiles.md:101]:
```suggestion
If someone wants to "lock for production" (meaning with production-related profiles), the answer is to keep the default profile and to use [releases](/docs/deployment/releases), which produce compiled artifacts that can be reused at any time.
```

I think this whole `x allows y-ing` is my broken English and there's generally a better way to go about it. No need to review it all, but I'll catch and suggest those I see as I go.

File: content/en/docs/configuration/profiles.md [hunk 3]
```diff
@@ -130,25 +129,21 @@ Rebar3 properly supports all these forms and merges them in a functional manner.
 ]}.
 ```
 
-Applying the profiles in various orders will yield different the following lists of options for `erl_opts`:
+Applying the profiles in various orders will differently yield the following lists of options for `erl_opts`:
```
Comment (ferd) [inline content/en/docs/configuration/profiles.md:132]:
```suggestion
Applying the profiles in various orders will yield different lists of options for `erl_opts`:
```

File: content/en/docs/tutorials/using_breakpoints_to_debug_tests.md [hunk 4]
```diff
@@ -214,54 +226,59 @@ start(_Config) ->
...
-	                     [gen_event]}}},
-	       undefined,0,1,[],0,break_check_sup,[]}
-	
-	%% and keep going to see the test pass
-	5> r3:resume().
-	ok
-	All 1 tests passed.
-	 
-It would be a good idea to remove all of your breakpoints after you're done with them, but it's interesting to know you can use features such as Common Test's [repeated executions with groups ](http://erlang.org/doc/apps/common_test/write_test_chapter.html#test-case-groups) or randomized runs in order to detect weird conditions and automatically trigger a break point when required.
+```shell
```
Comment (ferd) [inline content/en/docs/tutorials/using_breakpoints_to_debug_tests.md:249]: is shell or erlang a better language for these for syntax highlights?

**Review by ferd — CHANGES_REQUESTED** [review]
Thanks a lot! That's a huge, much needed clean up, and much appreciated.

I've included a few short changes (nothing major), and got a few questions about some of the syntax highlight language choices.

File: content/en/docs/configuration/dependencies.md (hunk 1 above)
Comment (paulo-ferraz-oliveira) [reply content/en/docs/configuration/dependencies.md:145]: Yeah, linter. I might remove those, sure. Not sure it's being highlighted, though. Could use ```plain, for example.

File: content/en/docs/tutorials/using_breakpoints_to_debug_tests.md (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply content/en/docs/tutorials/using_breakpoints_to_debug_tests.md:249]: I tried to choose based on the percentage of what I found most:
* `shell` should be used for e.g. bash stuff,
* `erlang` should be used for actual Erlang code.

I'll re-review all of these, too. I got into the `markdownlint` craze and some might be off.

File: content/en/docs/configuration/dependencies.md (hunk 1 above)
Comment (ferd) [reply content/en/docs/configuration/dependencies.md:145]: yeah that'd be fine.

Comment (paulo-ferraz-oliveira) [reply content/en/docs/configuration/dependencies.md]: 511cdb9

File: content/en/docs/configuration/profiles.md (hunk 2 above)
Comment (paulo-ferraz-oliveira) [reply content/en/docs/configuration/profiles.md]: aae0f90

File: content/en/docs/tutorials/using_breakpoints_to_debug_tests.md (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply content/en/docs/tutorials/using_breakpoints_to_debug_tests.md]: All of these were review in a commit mentioned above.

**Review by ferd — APPROVED** [review]

**Pull request merged**